### ISSUE TITLE
Question: Installing from CRAN in pixi?

### ISSUE THREAD

**Issue by kaedonkers** [roaldarbol/rpix#11]
Hi roaldarbol,

Awesome project, much needed in the data science world!

I have been frustrated by the need to install R packages through a conda channel for a couple of years now. It feels like installing directly from CRAN would be a much more robust and clean solution, however I have never found anyone working on it. Given that this can be done with Python packages from pip (in either conda or pixi), I feel like it should be possible with R packages from CRAN.

Is this something you hope to achieve in this project? Do you have any thoughts on how installing directly from CRAN into a pixi environment (or indeed conda environment) might be done?

**Comment by roaldarbol** [comment]
Hey @kaedonkers!

Thanks for sharing the excitement! This package simply serves as an interface for pixi, so all the packaging is dealt with on the pixi side. But we are discussing all these things; I personally feel a robust conda packaging solution is needed, which the team is building in `rattler-build`. Alternatively they would have to work on something like porting renv. They did something similarlybfor pip, but since it was a lot of work and uv cane along it made more sense to utilize their work. 

It’s almost possible to package all the r-universe packages (which I believe includes all CRAN packages), it just need to be automated - however it’s their goal. You can follow the conversations here and give a thumbs up to the issue to let them know it should be prioritised.

{rpix} is simply going to be wrapping pixi CLI commands so we can install conda packages the way we R users normally install packages (in the R console), so there’s none of the “fun” stuff happening here, simply hoping to enable a good workflow akin to {rix} does for Nix.

**Comment by kaedonkers** [comment]
Great, thanks for the update! I agree that a conda-like solution for R packages is needed. I have tried `renv` but it doesn't quite deliver the polyglot solutions I need. I see you have blogged about using `renv`, any opinions on how that fits in?

Exciting to hear that the prefix team are working on automating the R-to-conda package pipeline, that will solve a lot!

**Comment by roaldarbol** [comment]
Yeah, it's the cross-language solutions I'm missing too. `renv` works well for R projects. I use it in pretty much all the projects I work on, and it has some nice features such as the one I blogged about. So yeah, great tool! For myself, I couldn't wait for a conda solution before getting on with my work, so I see it as an intermediate solution until we have great conda tooling. Consider joining their discord channel, would be great to have some more R voices to discuss with. :-)
